packages/core/src/main/Account.ts

```diff
@@ -616,19 +616,24 @@ export class Account<T = any> extends EventEmitter {
       // Lock websocket in order to buffer any message that arrives while we handle the notification stream
       onConnectionStateChanged(ConnectionState.PROCESSING_NOTIFICATIONS);
       this.apiClient.transport.ws.lock();
-      await this.service!.notification.handleNotificationStream(
+      const results = await this.service!.notification.processNotificationStream(
         async (notification, source, progress) => {
           await handleNotification(notification, source);
           onNotificationStreamProgress(progress);
         },
         onMissedNotifications,
         abortHandler,
       );
+      this.logger.log(`Finished processing notifications ${JSON.stringify(results)}`, results);
+      if (abortHandler.isAborted()) {
+        this.logger.warn('Ending connection process as websocket was closed');
+        return;
+      }
```

Comment:
The actual fix is only this, stopping the entire process when the aborthandler is actually aborted